init-common/functions

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+
+qsvc() {
+    # Returns whether a service is enabled.
+    # Usage: qsvc <nameofservice>
+    # Must only be used after qubes-sysinit has started.
+    test -e /var/run/qubes-service/"$1"
+}
+
+under_systemd() {
+    pidof systemd >/dev/null 2>&1
+}
+
+possibly_run_save_script()
+{
+    ENCODED_SCRIPT=$(qubesdb-read /qubes-save-script)
+    if [ -z "$ENCODED_SCRIPT" ] ; then return ; fi
+    tmpfile=$( mktemp qubes-save-script.XXXXXXXXX )
+    echo $ENCODED_SCRIPT|perl -e 'use MIME::Base64 qw(decode_base64); local($/) = undef;print decode_base64(<STDIN>)' >"$tmpfile"
+    chmod 755 "$tmpfile"
+    Xorg -config /etc/X11/xorg-preload-apps.conf :0 &
+    while ! [ -S /tmp/.X11-unix/X0 ]; do sleep 0.5; done
+    DISPLAY=:0 su - user -c "$tmpfile"
+    ret=$?
+    killall Xorg
+    rm -f "$tmpfile"
+    return $ret
+}
```

Comment:
This is wrong version of the function. It shouldn't use separate X server (indeed sysvinit version wasn't updated).
Calling this function needs to be ordered after qubes-gui-agent.service. It was before but just to be sure you haven't missed that.